File: scripts/algorithms/R/Reformat Phone Number/Reformat Phone Number.py
class Solution:
    def reformatNumber(self, number: str) -> str:
        s = number.replace(" ", "").replace("-", "")
        pieces = list()
        while s:
            if len(s) == 2:
                pieces.append(s)
                break
            elif len(s) == 4:
                pieces.append(s[:2])
                pieces.append(s[2:])
                break
            else:
                pieces.append(s[:3])
                s = s[3:]
        return "-".join(pieces)

File: scripts/algorithms/R/Reformat Phone Number/Reformat Phone Number.js
// Runtime: 86 ms (Top 60.00%) | Memory: 42.6 MB (Top 25.38%)
var reformatNumber = function(number) {
    let numArr = number.replace(/-/g,'').replace(/ /g,'').split('')
    let res = ''
    while(numArr.length >= 4){
        numArr.length == 4 ?
            res += numArr.splice(0,2).join('') +'-'+numArr.splice(0,2).join('') :
            res += numArr.splice(0,3).join('') + '-'
    }
    res += numArr.join('')
    return res
};